products/sle/main.pm

```diff
@@ -199,6 +199,39 @@ if (get_var('DEV_IMAGE')) {
     }
 }
 
+#This setting is used to enable all extensions/modules for upgrading tests, generate corresponding extensions/modules list for all addons, all extensions, all modules cases.
+if (check_var('SCC_REGISTER', 'installation') && get_var('ALL_ADDONS')) {
+    my $version = get_required_var("HDDVERSION");
+    #Below $common* store the extensions/modules list in common.
+    my $common_addons_12SP3     = 'sdk,pcm,tcm,wsm';
+    my $common_modules_12SP3    = 'pcm,tcm,wsm';
+    my $common_extensions_12SP3 = 'sdk';
+    my $common_addons_11SP4     = 'ha,geo,sdk';
```

Comment:
We got pretty complicated structure here,  which is hard to read if you see it for the first time. What's about putting those in the hashes which are initialized below depending on version? So that hash contains all addons/modules/extensions and we don't have to complement it later in the code.

Reply:
Agreeing with @rwx788 . Just as an example in case you do not know what we mean by hash:

Could look something like this

```perl
my %common_addons = (
    12SP3 => 'sdk,pcm,tcm,wsm',
    11SP4 => 'ha,geo,sdk'
);
my %common_modules = (
    12SP3 => 'pcm,tcm,wsm'
);
…
```